services/jvm/src/main/scala/io/gearpump/services/MasterService.scala

```diff
@@ -157,14 +152,46 @@ trait MasterService extends HttpService {
               PartitionerDescription(new PartitionerByClassName(edge))
             }
 
-            val appId = context.submit(new StreamApplication(appName, UserConfig.empty, graph))
+            val appId = context.submit(new StreamApplication(appName, userconfig, graph))
 
-            import upickle.default.{read, write}
+            import upickle.default.write
             val submitApplicationResultValue = SubmitApplicationResultValue(appId)
             val jsonData = write(submitApplicationResultValue)
             complete(jsonData)
           }
         }
+      } ~
+      path("uploadtasks") {
```

Comment:
Should this be more general uploadJar?

As the name uploadTasks doesn't make any sense, why a task can be uploaded?

I suggest use general UploadJar to do only one thing, which is to upload a jar file, this API will return the JarPath for this jar.

The frontend will be responsible to bind task class name to jarPath, and submit the DAG.